geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -1525,12 +1553,10 @@ Object nonTxnFindObject(KeyInfo keyInfo, boolean isCreate, boolean generateCallb
       }
       this.getFutures.remove(keyInfo.getKey());
```

Comment:
This finally block seems buggy in the case that "otherFuture" != null. In that case no other thread will see "thisFuture" because the putIfAbsent found "otherFuture". So calling thisFuture.set here is a waste of time.
But even worse, when we do the "this.getFutures.remove", we could be removing someone else's future from the map. I think it should use the flavor of ConcurrentMap remove here that says only remove if if it is equal to "thisFuture" and this whole finally block should be protected by "if (otherFuture == null".